zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/AQLGenerator.scala

```diff
@@ -20,9 +20,9 @@ class AQLGenerator extends IQLGenerator {
       case q: Query =>
         validateQuery(q)
         parseQuery(q, schemaMap)
-      case q: CreateView => parseCreate(q, schemaMap(q.query.dataset))
-      case q: AppendView => parseAppend(q, schemaMap(q.query.dataset))
-      case q: UpsertRecord => parseUpsert(q, schemaMap(q.dataset))
+      case q: CreateView => parseCreate(q, schemaMap(query.dataset))
+      case q: AppendView => parseAppend(q, schemaMap(query.dataset))
+      case q: UpsertRecord => parseUpsert(q, schemaMap(query.dataset))
```

Comment:
same here, `q.query` is the query that needs to run on the sourceDataset. what we need is the schema of the source not the target.